cluster-autoscaler/simulator/deprecated_scheduler_snapshot.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package simulator
```

Comment:
TODO: make sure this is actually deleted somewhere in this chain

Reply:
I will do that as last commit in the chain (maybe in separate PR) which will also address TODO is `PredicateChecker` interface.
I do not want to do it right now as this commit will with high probability conflict with any change on this PR. And I expect some changes more as we are during review.

Reply:
I will do that as last commit in the chain (maybe in separate PR) which will also address TODO is `PredicateChecker` interface.
I do not want to do it right now as this commit will with high probability conflict with any change on this PR. And I expect some changes more as we are during review.